File: sinq/src/test/java/models/Teacher_.java
package models;

import javax.persistence.metamodel.SetAttribute;
import javax.persistence.metamodel.SingularAttribute;
import javax.persistence.metamodel.StaticMetamodel;

/**
 * User: YaFengLi
 * Date: 12-12-14
 * Time: 下午5:36
 */
@StaticMetamodel(Teacher.class)
public class Teacher_ {
    public static volatile SingularAttribute<Teacher, Long> id;
    public static volatile SingularAttribute<Teacher, String> name;
    public static volatile SingularAttribute<Teacher, String> address;
    public static volatile SingularAttribute<Teacher, Integer> age;
    public static volatile SingularAttribute<Teacher, Husband> husband;
    public static volatile SetAttribute<Teacher, Student> students;
}

File: sinq/src/test/scala/models/Teacher.scala
package models

import java.util
import javax.persistence._

import scala.beans.BeanProperty

@Entity
@Table(name = "t_teacher")
case class Teacher(@BeanProperty var name: String, @BeanProperty var age: Int = -1, @BeanProperty var address: String) {
  @Id
  @GeneratedValue(strategy = GenerationType.TABLE, generator = "seq_t_teacher")
  @TableGenerator(name = "seq_t_teacher", table = "seq_t_teacher", allocationSize = 1)
  @BeanProperty
  var id: Long = _

  @OneToMany(cascade = Array(CascadeType.REMOVE), mappedBy = "teacher")
  @BeanProperty
  var students: util.Set[Student] = new util.HashSet[Student]()

  @OneToOne(cascade = Array(CascadeType.REMOVE), mappedBy = "teacher")
  @BeanProperty
  var husband: Husband = _

  def this() = this(null, -1, null)

}
